packages/vite/src/node/optimizer/index.ts

```diff
@@ -256,9 +256,15 @@ export async function optimizeDeps(
       const transformed = await transformWithEsbuild(entryContent, filePath, {
         loader: 'jsx'
       })
-      // Ensure that optimization won't fail.
-      esbuildOptions.loader ??= {}
-      esbuildOptions.loader['.js'] ??= 'jsx'
+      // Ensure that optimization won't fail by defaulting '.js' to the JSX parser.
+      // This is useful for packages such as Gatsby.
+      esbuildOptions = {
+        ...esbuildOptions,
```

Comment:
`esbuildOptions` is already local to this function, so it doesn't need to be cloned here